### ISSUE TITLE
Question about whitelisting

### ISSUE THREAD

**Issue by s0meguy1** [evilsocket/shieldwall#2]
Sorry to bother you again, but I am running a docker instance on the cloud server that is using shieldwall. I want to whitelist "172.17.0.1", but I'm not able to do so. In the config, it mentions lists of IP's to whitelist, do I add it like"

```
# list of ip addresses to always allow just in case
allow:
  - '127.0.0.1'
  - '172.17.0.1'
```

Or:

```
# list of ip addresses to always allow just in case
allow:
  - ['127.0.0.1', '172.17.0.1']
```

**Comment by evilsocket** [comment]
https://symfony.com/doc/current/components/yaml/yaml_format.html#collections

**Comment by evilsocket** [comment]
i'm working on docker compatibility as we speak, let me know if whitelisting that address is enough :D

**Comment by s0meguy1** [comment]
I don't know much about symfony/yaml and when I add a new line like:
  - '127.0.0.1'
  - '172.17.0.1'
I am getting:
```
2021-03-09 08:46:40 inf shieldwall agent v1.0.3
2021-03-09 08:46:56 inf using firewall /sbin/iptables
2021-03-09 08:46:56 !!! error decoding /var/lib/shieldwall/state.json: unexpected end of JSON input
```

**Comment by s0meguy1** [comment]
That worked perfectly! 

I am going to run a rest and make sure docker works, it should from the iptables output

**Comment by evilsocket** [comment]
let me push a workaround for the corrupted state file :D

**Comment by s0meguy1** [comment]
Hey was finally able to test, looks like something is failing, but I'm not sure if its on my end. In any case, here's the iptables -L -n output:

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
ACCEPT     tcp  --  X.X.X.86         0.0.0.0/0            tcp dpts:1:65535
ACCEPT     udp  --  X.X.X.86         0.0.0.0/0            udp dpts:1:65535
ACCEPT     tcp  --  127.0.0.1            0.0.0.0/0            tcp dpts:1:65535
ACCEPT     udp  --  127.0.0.1            0.0.0.0/0            udp dpts:1:65535
ACCEPT     tcp  --  172.17.0.1           0.0.0.0/0            tcp dpts:1:65535
ACCEPT     udp  --  172.17.0.1           0.0.0.0/0            udp dpts:1:65535
LOGNDROP   all  --  0.0.0.0/0            0.0.0.0/0

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain LOGNDROP (1 references)
target     prot opt source               destination
LOG        all  --  0.0.0.0/0            0.0.0.0/0            limit: avg 10/min burst 5 LOG flags 0 level 4 prefix "shieldwall-dropped: "
DROP       all  --  0.0.0.0/0            0.0.0.0/0
```


And the command to test if the docker image could reach google:

```
docker exec cd4c07f71f4f python -c "import requests; print(requests.get(url = 'https://google.com').text)"
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 169, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/lib/python3.9/socket.py", line 953, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Temporary failure in name resolution

During handling of the above exception, another exception occurred:
docker exec cd4c07f71f4f python -r "import requests; print(requests.get(url = 'https://google.com').text)"

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 353, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 181, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f8f83f10dc0>: Failed to establish a new connection: [Errno -  3] Temporary failure in name resolution

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='google.com', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('  <urllib3.connection.HTTPSConnection object at 0x7f8f83f10dc0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution  '))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/requests/api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='google.com', port=443): Max retries exceeded with url: / (Caused by NewConnectionErro  r('<urllib3.connection.HTTPSConnection object at 0x7f8f83f10dc0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolut  ion'))
```
I'll play around with it before work again tomorrow and confirm

**Comment by evilsocket** [comment]
nice! i was trying to reproduce it and that is the perfect one liner, thank you so much

**Comment by evilsocket** [comment]
for reference:

> docker run --rm nyurik/alpine-python3-requests python -c "import requests; print(requests.get(url = 'https://google.com').text)"

**Comment by evilsocket** [comment]
@s0meguy1 the fixes that @Matrix86 sent via PR successfully fixed the issue with docker, v1.1.0 now works like a charm without any specific configuration :D

**Comment by s0meguy1** [comment]
Awesome! Thank you for being so responsive!!
